### ISSUE TITLE
Attempting to Emit from an External Process, not working

### ISSUE THREAD

**Issue by EndenDragon** [miguelgrinberg/python-socketio#132]
I am working on a Discord bot in the backend that would emit events for the flask-socketio webapp. When the discord bot's on_message fires, I see the print `emitted` statement in the ssh terminal. However, I do not see the response in chrome's websockets Frames tab and not console logging the contents. I am attempting to use a redis server to bridge the connections.

bot.py
```
client = discord.Client()
external_sio = socketio.KombuManager('redis://', write_only=True)

@client.event
async def on_message(message):
    external_sio.emit('on_message', data={'content': message.content})
    print("emitted")
```

app.py - From the [example](https://github.com/miguelgrinberg/Flask-SocketIO/blob/master/example/app.py). Everything is kept relatively the same except this line:
`socketio = SocketIO(app, async_mode="gevent", message_queue='redis://')`

index.html - Also from the example with these lines added
```
socket.on('on_message', function(msg) {
    console.log(msg);
});
```

**Comment by miguelgrinberg** [comment]
The `KombuManager` class is not compatible with asyncio, you can use `AsyncRedisManager` instead.

**Comment by EndenDragon** [comment]
I switched it to AsyncRedisManager and added an await statement before it. It seems like nothing has changed (the issue is still present)

**Comment by miguelgrinberg** [comment]
Where is your code? Can I see it?

**Comment by EndenDragon** [comment]
This would be my app.py https://pastebin.com/0g5Kucrd

The following is my bot.py
```
import discord
import socketio

client = discord.Client()
external_sio = socketio.AsyncRedisManager('redis://', write_only=True)
@client.event
async def on_ready():
    print('Logged in as')
    print(client.user.name)
    print(client.user.id)
    print('------')

@client.event
async def on_message(message):
    await external_sio.emit('on_message', data={'content': message.content})
    print("emitted")

client.run('SHHH-SECRET!')
```

**Comment by miguelgrinberg** [comment]
The problem can be that your `app.py` defines the server to be on the `/test` namespace, but when you emit from the other process your are emitting on the default namespace. Try changing one of them to match the namespace of the other.

**Comment by EndenDragon** [comment]
Hm, adding `namespace='/test'` at the emit statement didn't seem to help either.
`await external_sio.emit('on_message', data={'content': message.content}, namespace='/test')`

**Comment by miguelgrinberg** [comment]
I can't run your app here since you didn't provide all the code for it, so the best I can do is find the problems one by one. You may want to add logging on client and server to see what's going on and where the message is getting stuck.

I assume your client is connecting to the `/test` namespace?

**Comment by EndenDragon** [comment]
I believe it is connecting to the /test namespace. Here is my full, uncut code in a temporary repository. https://github.com/EndenDragon/FlaskSocketIOTest/tree/master/example

**Comment by miguelgrinberg** [comment]
Okay, found a bunch more problems:

- You are using gevent, which does not support websocket natively. The gevent-websocket package does not run on Python3, so to I switched to eventlet, which has native support for websocket.
- You did not monkey patch the standard library for eventlet/gevent, which is required when using third party libraries that do sockets/threads/etc. I added this at the top of your `app.py` script.
- The `app.py` uses Flask-SocketIO, but the `bot.py` script uses python-socketio. These two have different defaults regarding the redis channel they use to communicate. Add `channel='flask-socketio'` to the constructor on `bot.py` to make the two use the same channel.

With all these changes, I get the message in the console window of the browser.
